Mypy doesn't detect unreachable if

* Are you reporting a bug, or opening a feature request?
Feature request (?)

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.
```python
class A:
    pass

if type(1) in (A,):
    print("this should give a warning")

if type(1) == A:
    print("this works")
```
* What is the actual behavior/output?
```
t.py:7: error: Non-overlapping equality check (left operand type: "Type[int]", right operand type: "Type[A]")
Found 1 error in 1 file (checked 1 source file)
```
* What is the behavior/output you expect?
Two errors, one for each impossible branch.
* What are the versions of mypy and Python you are using?
  Do you see the same issue after installing mypy from Git master?
Tried with `0.720`, `0.761` and `0.770+dev.974a58d3cb3029044b481f9e3154fa3a185cf0f3`
* What are the mypy flags you are using? (For example --strict-optional)
Tried with combinations of `--strict`, `--strict-equality`, and `--warn-unreachable`

--- a/mypy/checker.py
+++ b/mypy/checker.py
@@ -3461,6 +3461,8 @@ class TypeChecker(NodeVisitor[None], CheckerPluginInterface):
             super_instance = map_instance_to_supertype(typ, supertype)
             assert len(super_instance.args) == 1
             return super_instance.args[0]
+        if isinstance(typ, TupleType):
+            return self.analyze_container_item_type(tuple_fallback(typ))
         return None
 
     def analyze_index_variables(self, index: Expression, item_type: Type,

--- a/test-data/unit/check-expressions.test
+++ b/test-data/unit/check-expressions.test
@@ -2754,6 +2754,13 @@ assert u'foo' == 'foo'
 assert u'foo' == u'bar'  # E: Non-overlapping equality check (left operand type: "Literal[u'foo']", right operand type: "Literal[u'bar']")
 [builtins_py2 fixtures/python2.pyi]
 
+[case testStrictEqualityWithFixedLengthTupleInCheck]
+# flags: --strict-equality
+if 1 in ('x', 'y'):  # E: Non-overlapping container check (element type: "int", container item type: "str")
+    pass
+[builtins fixtures/tuple.pyi]
+[typing fixtures/typing-full.pyi]
+
 [case testUnimportedHintAny]
 def f(x: Any) -> None:  # E: Name 'Any' is not defined \
                         # N: Did you forget to import it from "typing"? (Suggestion: "from typing import Any")

